packages/components/src/components/as-category-widget/as-category-widget.tsx

```diff
@@ -0,0 +1,210 @@
+import { Component, Event, EventEmitter, Method, Prop, State } from '@stencil/core';
+import readableNumber from '../../utils/readable-number';
+import { shadeOrBlend } from '../../utils/styles';
+
+const OTHER_CATEGORY_COLOR = '#747474';
+const OTHER_CATEGORY_NAME = 'Other';
+const DEFAULT_BAR_COLOR = '#47DB99';
+
+@Component({
+  shadow: false,
+  styleUrl: './as-category-widget.scss',
+  tag: 'as-category-widget',
+})
+export class CategoryWidget {
+  @Prop() public categories: object[] = [];
+  @Prop() public defaultBarColor: string = DEFAULT_BAR_COLOR;
+  @Prop() public description: string;
+  @Prop() public heading: string;
+  @Prop() public showClearButton: boolean = false;
+  @Prop() public showHeader: boolean = true;
+  @Prop() public useTotalPercentage: boolean = false;
+  @Prop() public visibleCategories: number = Infinity;
+
+  @Event() public categoriesSelected: EventEmitter<string[]>;
+
+  @State() private selectedCategories: string[] = [];
+
+  @Method()
+  public getSelectedCategories() {
+    return this.selectedCategories;
+  }
+
+  @Method()
+  public clearSelection() {
+    if (!this.selectedCategories.length) {
+      return;
+    }
+
+    this.selectedCategories = [];
+    this._onCategoriesChanged();
+  }
+
+  public render() {
+    return [
+      this._renderHeader(),
+      this._renderCategoryList(),
+      this._renderFooter()
+    ];
+  }
+
+  private _renderHeader() {
+    if (!this.showHeader) {
+      return;
+    }
+
+    return [
+      <h2 class='as-subheader'>{this.heading}</h2>,
+      <p class='as-category-widget__description as-body'>{this.description}</p>,
+    ];
+  }
+
+  private _renderCategoryList() {
+    return <ul class='as-category-widget__list'>{this._renderCategories()}</ul>;
+  }
+
+  private _renderCategories() {
+    const moreCategoriesThanVisible = this.categories.length > this.visibleCategories;
+    const { categories, otherCategory } = this._parseCategories();
+    let otherCategoryTemplate;
+
+    const categoriesToRender =  moreCategoriesThanVisible
+      ? categories.slice(0, this.visibleCategories)
+      : categories;
+
+    const maximumValue = this.useTotalPercentage
+      ? this._getCategoriesTotalValue(this.categories)
+      : this._getVisibleCategoriesMaximumValue();
+
+    if (otherCategory || moreCategoriesThanVisible) {
+      otherCategoryTemplate = this._renderOtherCategory(otherCategory, { maximumValue });
+    }
+
+    return [
+      categoriesToRender.map((category: Category) => this._renderCategory(category, { maximumValue })),
+      otherCategoryTemplate
+    ];
+  }
+
+  private _renderCategory(category: Category, options: CategoryOptions) {
+    const isSelected = this._isSelected(category.name);
+    const isAnyCategorySelected = this.selectedCategories.length > 0;
+    const barColor = options.isOther
```

Comment:
Can't we move the `isOther` logic to the `getBarColor` method? In my head the `renderCategory`method shouldn't do this work, I expect `getBarColor` to be smart enough to give me the final color, also, we are doing something similar with the `isSelected` what do you think?

Reply:
Yes, it would be so much better :)